atom/browser/api/atom_api_window.cc

```diff
@@ -914,6 +915,65 @@ void Window::ResetBrowserView() {
   browser_view_.Reset();
 }
 
+void Window::AddBrowserView(v8::Local<v8::Value> value) {
+  mate::Handle<BrowserView> browser_view;
+  if (value->IsNull() || value->IsUndefined()) {
+  } else if (mate::ConvertFromV8(isolate(), value, &browser_view)) {
+    auto get_that_view = browser_views_.find(browser_view->weak_map_id());
+    if (get_that_view == browser_views_.end()) {
+      window_->AddBrowserView(browser_view->view());
+      browser_view->web_contents()->SetOwnerWindow(window_.get());
+      browser_views_[browser_view->weak_map_id()].Reset(isolate(), value);
+    }
+    window_->UpdateDraggableRegionViews();
+  }
+}
+
+void Window::RemoveBrowserView(v8::Local<v8::Value> value) {
+  mate::Handle<BrowserView> browser_view;
+  if (value->IsNull() || value->IsUndefined()) {
+  } else if (mate::ConvertFromV8(isolate(), value, &browser_view)) {
+    auto get_that_view = browser_views_.find(browser_view->weak_map_id());
+    if (get_that_view != browser_views_.end()) {
+      window_->RemoveBrowserView(browser_view->view());
+      browser_view->web_contents()->SetOwnerWindow(nullptr);
+
+      (*get_that_view).second.Reset(isolate(), value);
+      browser_views_.erase(get_that_view);
+    }
+    window_->UpdateDraggableRegionViews();
+  }
+}
+
+void Window::ResetBrowserViews() {
+  for (auto views_iter = browser_views_.begin();
```

Comment:
Can use range-based for loop here:

```c++
for (auto item : browser_views_) {
  item.second.Reset()
}
```

Other for loops can also be modified as such.

Reply:
done.